src/DefinitionResolver.php

```diff
@@ -144,6 +150,10 @@ public function resolveReferenceNodeToDefinition(Node $node)
     {
         // Variables are not indexed globally, as they stay in the file scope anyway
         if ($node instanceof Node\Expr\Variable) {
+            // Resolve $this
+            if ($node->name === 'this' && $fqn = $this->getContainingClassFqn($node)) {
+                return $this->index->getDefinition($fqn, false);
+            }
```

Comment:
Okay. If you do it in a separate PR I think I would merge it straight away, change looks good.